### PR TITLE
chore: refactoring for network request intercept

### PR DESCRIPTION
**Pull request by thiagowfx**
Bug: #644

### CODE REVIEW

**Comment by thiagowfx** [comment]
Forked from https://github.com/GoogleChromeLabs/chromium-bidi/pull/957

File: src/bidiMapper/domains/context/browsingContextProcessor.ts [hunk 1]
```diff
@@ -570,10 +570,7 @@ export class BrowsingContextProcessor {
     const context = params.context;
     const sessionId =
       this.#browsingContextStorage.getContext(context).cdpTarget.cdpSessionId;
-    if (sessionId === undefined) {
-      return {result: {session: null}};
-    }
-    return {result: {session: sessionId}};
+    return {result: {session: sessionId === undefined ? null : sessionId}};
```
Comment (Lightning00Blade) [inline src/bidiMapper/domains/context/browsingContextProcessor.ts:573]:
```suggestion
    return {result: {session: sessionId ? null : sessionId}};
```

File: src/bidiMapper/domains/context/cdpTarget.ts [hunk 2]
```diff
@@ -36,8 +36,8 @@ export class CdpTarget {
   readonly #eventManager: IEventManager;
   readonly #preloadScriptStorage: PreloadScriptStorage;
 
-  readonly #targetUnblocked: Deferred<void>;
-  #networkDomainActivated: boolean;
+  readonly #targetUnblocked = new Deferred<void>();
```
Comment (Lightning00Blade) [inline src/bidiMapper/domains/context/cdpTarget.ts:39]: Thanks was going to open PR for this myself.

File: src/bidiMapper/domains/context/browsingContextProcessor.ts (hunk 1 above)
Comment (thiagowfx) [reply src/bidiMapper/domains/context/browsingContextProcessor.ts:573]: Are you sure this is a no-op?

**Review by Lightning00Blade — COMMENTED** [review]
LGTM,  but keep in mind we are have draft PR to move away from the manual types so this work will be remove in the following days any way.

**Review by Lightning00Blade — APPROVED** [review]

**Pull request merged**